File: 118a.exs
defmodule Main do
  def remove_vowels(vowels, word, n, new_word) do
    if n >= String.length(word) - 1 do
      new_word
    else
      current = String.at(word, n) |> String.downcase()

      if String.capitalize(current) in vowels do
        remove_vowels(vowels, word, n + 1, new_word)
      else
        remove_vowels(vowels, word, n + 1, new_word <> "." <> current)
      end
    end
  end

  def main do
    word = IO.gets("")
    vowels = ["A", "O", "Y", "E", "U", "I"]
    new_word = remove_vowels(vowels, word, 0, "")
    IO.puts(new_word)
  end
end

Main.main()

File: 118a.ml
let word = read_line ()
let vowels = [ "A"; "O"; "Y"; "E"; "U"; "I" ]

let rec exists c l =
  match l with [] -> false | h :: t -> if h = c then true else exists c t

let rec remove_vowels vowels word n new_word =
  if n >= String.length word then new_word
  else
    (* NOTE: use `lowercase` and `capitalize` when submitting to cf *)
    let current = word.[n] |> String.make 1 |> String.lowercase_ascii in
    if exists (String.capitalize_ascii current) vowels then
      remove_vowels vowels word (n + 1) new_word
    else remove_vowels vowels word (n + 1) (new_word ^ "." ^ current)

let () = print_string (remove_vowels vowels word 0 "")
